### DISCUSSION TITLE
Looking for help with SAS commands or version

### DISCUSSION THREAD

**Discussion by JonhDoe1998** [zacharytomlinson/saspy#39] [Q&A]
Hi, recently I get a Aristocart Viridian Gen 7 model slot machine(what they told), and I am trying to write some Host to control the machine with a Raspberry Pi. 

Well the connection thing is okay, and some of the R poll command in saspy is properly responded by the machine, still there are some not get any response from the machine, especially the AFT related commands.

After checking the IGT's SAS protocol doc, now get confused, is it the protocol version problem or the machine configuration problem, can not figure it out. Anyone can help with this?

**Comment by JonhDoe1998** [comment]
sorry guys I have found my mistake in code.

**Reply by well-it-wasnt-me** [reply]
Any chance you can share what went wrong ? Might be helpful for someone who would make your same mistake

**Reply by JonhDoe1998** [reply]
Ah yes, of course, but it's quite a stupid mistake. At first I am not quite understand the 'wake up bit' thing(noob to hardware communication), so I changed the line which set parity value try to find out the difference, but then interrupted by something else, and when I come back to this, I totally forgot that I have changed the code, until I checked the unstaged changes of git...

I tried ODD and EVEN, that's why some R poll works and some not..and why aft command totally not work.

But yes I also has to change some code in the aft_in to make it work, because according to the protocol, the byte after 0x72 should be the length of bytes after, I don't know why you guys set it with the Key thing, but after I change it to length, it works on my machine now.
